File: ejemplo_clase/vehiculo.cpp
#include "vehiculo.h"

int id_aux = 0;

Vehiculo::Vehiculo(){
  
  this->estado_motor = false;
  this->marcha = 0;
  this->auto_id = ++id_aux;
}

Vehiculo::Vehiculo(string modelo, string tipo_motor){
  this->estado_motor = false;
  this->marcha = 0;
  this->auto_id = ++id_aux;
  
  this->modelo = modelo;
  this->tipo_motor = tipo_motor;
}

Vehiculo::~Vehiculo(){
  cout<< "Auto " << this->auto_id << " Destructor" << endl;
}

void Vehiculo::encender(){
  if( !this->estado_motor ){
    this->estado_motor = true;
  }
  cout << "Motor encendido" << endl;
}

int Vehiculo::aumentar_marcha(){
  ++(this->marcha);
  return this->marcha;
}

void Vehiculo::acelerar(int v){
  if(this->velocidad < v){
    this->velocidad = v;
  }
}

File: clase06/oop/Vehiculo.java
package clase06.oop;

public class Vehiculo implements IVehiculo {
	
	public String Modelo;
	public boolean EstadoMotor;
	public String TipoMotor;
	public int Marcha;
	public int Velocidad;

	public Vehiculo() {

	}

	public Vehiculo(String modelo, String tipo_motor) {
		this.Modelo = modelo;
		this.TipoMotor = tipo_motor;
		this.Velocidad = 0;
		this.Marcha = 0;
	}

	public void Encender() {
		if (!this.EstadoMotor) {
			this.EstadoMotor = true;
		}

		System.out.println("Motor encendido");
	};

	public int AumentarMarcha() {
		++(this.Marcha);
		return this.Marcha;
	};

	public void Acelerar(int v) {
		if (this.Velocidad < v) {
			this.Velocidad = v;
		}
	};
}